src/main/java/net/snowflake/client/core/auth/AuthenticatorType.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing Inc. All right reserved.
+ */
+package net.snowflake.client.core.auth;
+
+import net.snowflake.client.core.SnowflakeJdbcInternalApi;
+
+@SnowflakeJdbcInternalApi
+public enum AuthenticatorType {
+  /*
+   * regular login username+password via Snowflake, may or may not have MFA
+   */
+  SNOWFLAKE,
+
+  /*
+   * federated authentication, OKTA as IDP
+   */
+  OKTA,
+
+  /*
+   * Web browser based authenticator for SAML 2.0 compliant
+   * service/application
+   */
+  EXTERNALBROWSER,
+
+  /*
+   * OAUTH 2.0 flow
+   */
+  OAUTH,
+
+  /*
+   * Snowflake local authentication using jwt token as a user credential
+   */
+  SNOWFLAKE_JWT,
+
+  /*
+   * Internal authenticator to enable id_token for web browser based authenticator
+   */
+  ID_TOKEN,
+
+  /*
+   * Authenticator to enable token for regular login with mfa
+   */
+  USERNAME_PASSWORD_MFA,
+
+  /*
+   * Authenticator to support PAT
+   */
+  PROGRAMMATIC_ACCESS_TOKEN,
```

Comment:
Do we want to add it now or maybe when it is actually needed?

Reply:
It's already needed :)

Reply:
As discussed I can add it in the next PR when it will actually be used.